test/core/tsi/ssl_transport_security_utils_test.cc

```diff
@@ -429,6 +447,182 @@ INSTANTIATE_TEST_SUITE_P(FrameProtectorUtil, FlowTest,
+    ASSERT_EQ(evil_ca.status(), absl::OkStatus());
+    evil_ca_ = ReadPemCert(evil_ca->as_string_view());
+  }
+
+  void TearDown() override {
+    X509_CRL_free(root_crl_);
+    X509_CRL_free(intermediate_crl_);
+    X509_CRL_free(invalid_signature_crl_);
+    X509_free(root_ca_);
+    X509_free(intermediate_ca_);
+    X509_free(leaf_cert_);
+    X509_free(evil_ca_);
+  }
+
+ protected:
+  X509_CRL* root_crl_;
+  X509_CRL* intermediate_crl_;
+  X509_CRL* invalid_signature_crl_;
+  X509* root_ca_;
+  X509* intermediate_ca_;
+  X509* leaf_cert_;
+  X509* evil_ca_;
+};
+
+TEST_F(CrlUtils, VerifySignatureValid) {
+  EXPECT_TRUE(VerifyCrlSignature(root_crl_, root_ca_));
+}
+
+TEST_F(CrlUtils, VerifySignatureIntermediateValid) {
+  EXPECT_TRUE(VerifyCrlSignature(intermediate_crl_, intermediate_ca_));
+}
+
+TEST_F(CrlUtils, VerifySignatureModifiedSignature) {
+  EXPECT_FALSE(VerifyCrlSignature(invalid_signature_crl_, root_ca_));
+}
+
+TEST_F(CrlUtils, VerifySignatureModifiedContent) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kModifiedContent, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  X509_CRL* crl = ReadCrl(crl_slice->as_string_view());
+  EXPECT_EQ(crl, nullptr);
+}
+
+TEST_F(CrlUtils, VerifySignatureWrongIssuer) {
+  EXPECT_FALSE(VerifyCrlSignature(root_crl_, intermediate_ca_));
+}
+
+TEST_F(CrlUtils, VerifySignatureWrongIssuer2) {
+  EXPECT_FALSE(VerifyCrlSignature(intermediate_crl_, root_ca_));
+}
+
+TEST_F(CrlUtils, VerifySignatureNullCrl) {
+  EXPECT_FALSE(VerifyCrlSignature(nullptr, root_ca_));
+}
+
+TEST_F(CrlUtils, VerifySignatureNullCert) {
+  EXPECT_FALSE(VerifyCrlSignature(intermediate_crl_, nullptr));
+}
+
+TEST_F(CrlUtils, VerifySignatureNullCrlAndCert) {
+  EXPECT_FALSE(VerifyCrlSignature(nullptr, nullptr));
+}
+
+TEST_F(CrlUtils, VerifyIssuerNamesMatch) {
+  EXPECT_TRUE(VerifyCrlCertIssuerNamesMatch(root_crl_, root_ca_));
+}
+
+TEST_F(CrlUtils, VerifyIssuerNamesDontMatch) {
+  EXPECT_FALSE(VerifyCrlCertIssuerNamesMatch(root_crl_, leaf_cert_));
+}
+
+TEST_F(CrlUtils, DuplicatedIssuerNamePassesButSignatureCheckFails) {
+  // The issuer names will match, but it should fail a signature check
+  EXPECT_TRUE(VerifyCrlCertIssuerNamesMatch(root_crl_, evil_ca_));
+  EXPECT_FALSE(VerifyCrlSignature(root_crl_, evil_ca_));
+}
+
+TEST_F(CrlUtils, VerifyIssuerNameNullCrl) {
+  EXPECT_FALSE(VerifyCrlCertIssuerNamesMatch(nullptr, root_ca_));
+}
+
+TEST_F(CrlUtils, VerifyIssuerNameNullCert) {
+  EXPECT_FALSE(VerifyCrlCertIssuerNamesMatch(intermediate_crl_, nullptr));
+}
+
+TEST_F(CrlUtils, VerifyIssuerNameNullCrlAndCert) {
+  EXPECT_FALSE(VerifyCrlCertIssuerNamesMatch(nullptr, nullptr));
+}
+
+TEST_F(CrlUtils, HasCrlSignBitExists) { EXPECT_TRUE(HasCrlSignBit(root_ca_)); }
+
+// On OpenSSL 1.0.2 this would be expected to just pass all the time because
+// X509_get_key_usage doesn't exist, so we make it a passthrough
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
```

Comment:
as above, I think you can fix this and then make this test run unconditionally.

Reply:
Removed the include guards and comments